sway-core/src/transform/to_parsed_lang/convert_parse_tree.rs

```diff
@@ -97,6 +104,7 @@ fn item_to_ast_nodes(
     engines: Engines<'_>,
     item: Item,
     is_root: bool,
+    is_inside_code_block: bool,
```

Comment:
I prefer something like `is_associated_const_in_trait` which is only `true` when the `const` is an associated constant in a trait definition, and `false` otherwise.